File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.cpp
class Solution {
public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        unordered_map<int,int>mp; 

        for( auto num : nums){
            mp[num]++; 
        }

        vector<int>ans; 
        for( auto num : nums){
            if(mp[num] == 2) {
                ans.push_back(num);
                mp[num] =1;
            } 
        }
        return ans ;
    }
};

File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
          HashMap<Integer, Integer> mp = new HashMap<>();

        // Count occurrences
        for (int num : nums) {
            mp.put(num, mp.getOrDefault(num, 0) + 1);
        }

        List<Integer> ans = new ArrayList<>();
        for (int num : nums) {
            if (mp.get(num) == 2) {
                ans.add(num);
                mp.put(num, 0); // So it's not added again
            }
        }

        // Convert List<Integer> to int[]
        int[] result = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++) {
            result[i] = ans.get(i);
        }

        return result;
        
    }
}